UefiPayloadPkg/Library/FdtParserLib/FdtParserLib.c

```diff
@@ -282,6 +290,15 @@ ParseReservedMemory (
           PlatformAcpiTable->Header.Revision = UNIVERSAL_PAYLOAD_ACPI_TABLE_REVISION;
           PlatformAcpiTable->Header.Length   = sizeof (UNIVERSAL_PAYLOAD_ACPI_TABLE);
         }
+      } else if (AsciiStrnCmp (TempStr, "smbios", AsciiStrLen ("smbios")) == 0) {
+        DEBUG ((DEBUG_INFO, " build smbios, NumberOfBytes:%x", NumberOfBytes));
+        BuildMemoryAllocationHob (StartAddress, NumberOfBytes, EfiBootServicesData);
+        SmbiosTable = BuildGuidHob (&gUniversalPayloadSmbios3TableGuid, sizeof (UNIVERSAL_PAYLOAD_SMBIOS_TABLE));

```

Comment:
instead of hardcoding value, we should use NumberOfBytes reported by FDT node to BuildGuidHob so it will reserve the whole data buffer.